pandas/tests/indexes/test_interval.py

```diff
@@ -348,6 +348,109 @@ def test_get_loc_value(self):
         idx = IntervalIndex.from_arrays([0, 2], [1, 3])
         pytest.raises(KeyError, idx.get_loc, 1.5)
 
+    @pytest.mark.xfail(reason="new indexing tests for issue 16316")
+    def test_get_loc_value(self):
+
+        right = IntervalIndex.from_tuples([(0, 1), (2, 3)], closed='right')
+
+        pytest.raises(KeyError, right.get_loc(-0.5) )
+        pytest.raises(KeyError, right.get_loc(0) )
+        assert                  right.get_loc(0.5) == 0
+        assert                  right.get_loc(1)   == 0
+        pytest.raises(KeyError, right.get_loc(1.5) )
+        pytest.raises(KeyError, right.get_loc(2) )
+        assert                  right.get_loc(2.5) == 1
+        assert                  right.get_loc(3)   == 1
+        pytest.raises(KeyError, right.get_loc(3.5) )
+
+        assert                  right.get_loc(Interval(0, 1, closed='right')) == 0
+        pytest.raises(KeyError, right.get_loc(Interval(1, 2, closed='right')) )
+        assert                  right.get_loc(Interval(2, 3, closed='right')) == 1
+        pytest.raises(KeyError, right.get_loc(Interval(3, 4, closed='right')) )
+        pytest.raises(KeyError, right.get_loc(Interval(0, 2, closed='right')) )
+        pytest.raises(KeyError, right.get_loc(Interval(2.5, 3, closed='right')) )
+
+        pytest.raises(KeyError, right.get_loc(Interval(0, 1, closed='left')) )
+        pytest.raises(KeyError, right.get_loc(Interval(1, 2, closed='left')) )
+        pytest.raises(KeyError, right.get_loc(Interval(2, 3, closed='left')) )
+        pytest.raises(KeyError, right.get_loc(Interval(3, 4, closed='left')) )
+        pytest.raises(KeyError, right.get_loc(Interval(0, 2, closed='left')) )
+        pytest.raises(KeyError, right.get_loc(Interval(2.5, 3, closed='left')) )
+
+        pytest.raises(KeyError, right.get_loc(Interval(0, 1, closed='both')) )
+        pytest.raises(KeyError, right.get_loc(Interval(1, 2, closed='both')) )
+        pytest.raises(KeyError, right.get_loc(Interval(2, 3, closed='both')) )
+        pytest.raises(KeyError, right.get_loc(Interval(3, 4, closed='both')) )
+        pytest.raises(KeyError, right.get_loc(Interval(0, 2, closed='both')) )
+        pytest.raises(KeyError, right.get_loc(Interval(2.5, 3, closed='both')) )
+
+        left = IntervalIndex.from_tuples([(0, 1), (2, 3)], closed='left')
```

Comment:
This is long enough for one test. Can you make this a second method, e.g., `test_get_loc_vlaue_closed_left`?